### PR TITLE
ref(hybrid): Slugless route paths

### PR DESCRIPTION
**Pull request by mikejihbe**
Add a helper that retains legacy routepaths containing org slugs but adds additional mirrored paths without those slugs and redirects to them if applicable.

### CODE REVIEW

File: static/app/routes.tsx [hunk 1]
```diff
@@ -67,6 +67,269 @@ export function makeLazyloadComponent<C extends ComponentType>(
 // Shorthand to avoid extra line wrapping
 const make = makeLazyloadComponent;
 
+const _ = [
```
Comment (mikejihbe) [inline static/app/routes.tsx:70]: Don't fret, this will be removed, it's just for reference

File: static/app/routes.tsx [hunk 2]
```diff
@@ -1890,9 +2209,64 @@ function buildRoutes() {
     </Route>
   );
 
+  let paths = [];
```
Comment (mikejihbe) [inline static/app/routes.tsx:2212]: This is all also just for debugging

File: static/app/routes.tsx [hunk 3]
```diff
@@ -148,6 +411,43 @@ function buildRoutes() {
     </Route>
   ) : null;
 
+  /*
```
Comment (mikejihbe) [inline static/app/routes.tsx:414]: This is the part to comment on, and it's example usage below

File: static/app/routes.tsx [hunk 4]
```diff
@@ -148,6 +411,43 @@ function buildRoutes() {
...
+    sentry.io/settings/{slug}/ -> {slug}.sentry.io/settings/
+    sentry.io/onboarding/{slug}/{step} -> {slug}.sentry.io/onboarding/{step}
+    sentry.io/settings/{slug}/projects/{projectslug}/ -> {slug}.sentry.io/settings/projects/{project.slug}/
+  */
+  const orgScopedCustomDomainRoute = originalRoute => {
+    const newPath = originalRoute.props.path
+      .replace(/organizations\/:orgId\/?/, '')
+      .replace(/:orgId\/?/, '');
+    const customDomainRoute = (
+      <Route path={newPath} component={originalRoute.component} />
```
Comment (markstory) [inline static/app/routes.tsx:429]: We will need to copy `props` from the existing routes too. Several issue related routes have `props` defined.

We might also need a branch to handle `IndexRoute` instances and existing `Redirect` routes.

Comment (mikejihbe) [reply static/app/routes.tsx]: I revamped this. I ended up having to put together a full tree traversal. It's not quite working yet, but I think it's pretty close

**Comment by mikejihbe** [comment]
Subsumed by other work

**Pull request closed without merging**